controller/cache/cache.go

```diff
@@ -394,6 +395,15 @@ func (c *liveStateCache) getCluster(server string) (clustercache.ClusterCache, e
 		return nil, fmt.Errorf("error getting custom label: %w", err)
 	}
 
+	// controller dynamically fetches all resource types available on the cluster using a discovery API
+	// that may contain deprecated APIs. This causes log flooding when managing a large number of clusters.
+	// https://github.com/argoproj/argo-cd/issues/11973
+	// However, we can safely suppress deprecation warnings
+	// because we do not rely on resources with a particular API group or version.
+	// https://kubernetes.io/blog/2020/09/03/warnings/#customize-client-handling
+	clusterCacheConfig := cluster.RESTConfig()
+	clusterCacheConfig.WarningHandler = rest.NoWarnings{}
```

Comment:
Hmm, I'm not sure if I agree this should be completely shut off. Perhaps if this is wrapped like this, where it is enabled only for log levels that are higher than Debug?

```suggestion
	if log.GetLevel() >= log.DebugLevel {
		clusterCacheConfig.WarningHandler = rest.NoWarnings{}
	}
```

Reply:
Yeah, it can be helpful for debugging purposes. I'm fine with leaving the option to log the warnings.
@sidewinder12s, @jannfis WDYT?

Reply:
Agreed it should just be hidden in higher log levels.

Reply:
Thanks, I have revised the commit to reflect the suggestion.